distsys/archetyperesource.go

```diff
@@ -137,6 +148,27 @@ func (res *LocalArchetypeResource) Close() error {
 	return nil
 }
 
+func (res *LocalArchetypeResource) ForkState() (ArchetypeResource, error) {
+	return &LocalArchetypeResource{
+		value:       res.value,
+		oldValue:    res.oldValue,
+		hasOldValue: res.hasOldValue,
+		forkParent:  res,
+	}, nil
+}
+
+func (res *LocalArchetypeResource) LinkState() error {
+	res.forkParent.value = res.value
```

Comment:
following on for why this method should not exist imo, notice how there would be literally no difference if you just kept one of the forked resources, vs doing a field by field full copy here.